Meta/CMake/code_generators.cmake

```diff
@@ -32,6 +32,10 @@ function(compile_ipc source output)
     get_filename_component(output_name ${output} NAME)
     add_custom_target(generate_${output_name} DEPENDS ${output})
     add_dependencies(all_generated generate_${output_name})
+
+    string(LENGTH ${SerenityOS_SOURCE_DIR} root_source_dir_length)
+    string(SUBSTRING ${CMAKE_CURRENT_SOURCE_DIR} ${root_source_dir_length} -1 current_source_dir_relative)
```

Comment:
Hmm. Could you add a to-do to use `cmake_path` when we bump the CMake minimum required up to 3.20+? https://cmake.org/cmake/help/v3.23/command/cmake_path.html#relative-path . Looks fine otherwise 🙂

Reply:
Sure, would defn be nice to use that when we upgrade. I also added a TODO about that in another place.